packages/fxa-auth-server/lib/routes/utils/subscriptions.js

```diff
@@ -29,43 +29,131 @@ module.exports = {
+        uid,
+        stripeHelper,
+        email
+      );
+      await checkCapabilitiesFromStripe(
+        subscribedProducts,
+        capabilitiesToReveal,
+        client_id,
+        stripeHelper
+      );
+    } else {
+      // TODO: issue #3846 - remove this conditional branch
+      await fetchSubscribedProductsFromLocalDB(subscribedProducts, db, uid);
+    }
+
     // Accounts with at least one subscription get this product
-    if (subscriptions.length > 0) {
+    if (subscribedProducts.length > 1) {
       subscribedProducts.push(PRODUCT_SUBSCRIBED);
     }
 
-    const subscribedCapabilities = subscribedProducts.reduce(
-      (capabilities, product) =>
-        capabilities.concat(productCapabilities[product] || []),
-      []
-    );
-
-    const clientVisibleCapabilities = clientCapabilities[client_id] || [];
-
-    const capabilitiesToReveal = new Set(
-      subscribedCapabilities.filter(
-        capability =>
-          auth.strategy === 'sessionToken' ||
-          clientVisibleCapabilities.includes(capability)
-      )
+    // TODO: issue #3913 - remove support for capabilities from server config
+    checkCapabilitiesFromServerConfig(
+      capabilitiesToReveal,
+      subscribedProducts,
+      client_id,
+      config,
+      auth
     );
 
     return capabilitiesToReveal.size > 0
       ? Array.from(capabilitiesToReveal)
       : undefined;
   },
-};
+});
+
+async function fetchSubscribedProductsFromStripe(
+  subscribedProducts,
+  uid,
+  stripeHelper,
+  email
+) {
+  const customer = await stripeHelper.customer(uid, email);
+  if (!customer || !customer.subscriptions.data) {
+    return;
+  }
+  subscribedProducts.push(
+    ...customer.subscriptions.data.map(
+      ({ plan: { product: productId } }) => productId
+    )
+  );
+}
+
+async function checkCapabilitiesFromStripe(
+  subscribedProducts,
+  capabilitiesToReveal,
+  client_id,
+  stripeHelper
+) {
+  // Run through all plans and collect capabilitiies for subscribed products
+  const plans = await stripeHelper.allPlans();
+  for (const plan of plans) {
+    if (!subscribedProducts.includes(plan.product_id)) {
+      continue;
+    }
+    const metadata = SubscriptionUtils.metadataFromPlan(plan);
+    const capabilityKeys = [
+      'capabilities',
+      ...Object.keys(metadata).filter(key =>
+        client_id === null
+          ? key.startsWith('capabilities:')
+          : key === `capabilities:${client_id}`
+      ),
+    ].filter(key => key in metadata);
+    for (const key of capabilityKeys) {
+      const capabilities = metadata[key].split(',');
+      for (const capability of capabilities) {
+        capabilitiesToReveal.add(capability);
+      }
+    }
+  }
+}
+
+// TODO: issue #3846 - remove this method
+async function fetchSubscribedProductsFromLocalDB(subscribedProducts, db, uid) {
+  const subscriptions = (await db.fetchAccountSubscriptions(uid)) || [];
+  subscribedProducts.push(...subscriptions.map(({ productId }) => productId));
```

Comment:
`concat` instead of `push` spreaded new array?

Reply:
Concat creates a new array, whereas I'm adding to an existing array here

Reply:
Ah, true true 👍